### PR TITLE
fix(deploy): update DaemonSet config to deploy tracee in K8s

### PR DESCRIPTION
**Pull request by danielpacak**
Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: deploy/kubernetes/tracee-falcosidekick/tracee.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ spec:
       name: tracee
     spec:
       containers:
-      - image: aquasec/tracee:latest
-        imagePullPolicy: Always
+      - image: docker.io/aquasec/tracee:v0.7.0-rc-1
+        imagePullPolicy: IfNotPresent
         args:
           - --webhook http://tracee-webhook:2801 --webhook-template ./templates/falcosidekick.tmpl --webhook-content-type application/json
```
Comment (danielpacak) [inline deploy/kubernetes/tracee-falcosidekick/tracee.yaml:23]: Note that due to #1577 the Pod will enter the crash loop.

File: Vagrantfile [hunk 2]
```diff
@@ -9,6 +9,18 @@ Vagrant.configure("2") do |config|
   # Forward MkDocs dev server to preview documentation on the host at http://localhost:8000/tracee
   config.vm.network :forwarded_port, guest: 8000, host: 8000
 
+  # Forward MicroK8s dashboard to access it on the host at https://localhost:10443
```
Comment (danielpacak) [inline Vagrantfile:12]: tracee_on_k8s

File: deploy/kubernetes/tracee-falcosidekick/tracee.yaml [hunk 3]
```diff
@@ -27,11 +27,8 @@ spec:
         volumeMounts:
         - name: tmp-tracee
           mountPath: /tmp/tracee
-        - name: usr-src
```
Comment (danielpacak) [inline deploy/kubernetes/tracee-falcosidekick/tracee.yaml:30]: I wasn't sure if we still need these volume mounts?

**Review by itaysk — APPROVED** [review]
LGTM
about the docker image versions - sure we want to change latest to pinned version and maintain it?
Also, just a reminder that the `v` in the image name needs to be fixed either before this PR is merged or with the same PR that fixes it.

**Comment by danielpacak** [comment]
> LGTM
> about the docker image versions - sure we want to change latest to pinned version and maintain it?
> Also, just a reminder that the `v` in the image tag needs to be fixed either before this PR is merged or with the same PR that fixes it.

IMHO using pinned tag instead of latest adds some effort to maintain it, but at least we won't break installation experience for previous versions of tracee. For example, if next time we publish the latest image that requires more environment variables to be set, but previous deployment YAML doesn't set these envs tracee may not even star.

Regarding the v prefix we need to figure out if v0.7.0-rc-1 was a bug or a feature. I'll open another PR in any case to make sure we use the valid image tag. /cc @grantseltzer

**Comment by itaysk** [comment]
> Regarding the v prefix we need to figure out if v0.7.0-rc-1 was a bug or a feature.

I think in docker it's conveniently omitted. Not sure why though

**Comment by danielpacak** [comment]
> > Regarding the v prefix we need to figure out if v0.7.0-rc-1 was a bug or a feature.
> 
> I think in docker it's conveniently omitted. Not sure why though

We've merged #1586 . So I'm going to open another PR with image tags without the "v" prefix.

**Pull request merged**